### ISSUE TITLE
MD057: trailing-slash absolute directory link reports a misleading "not found", and requires index.md outside a docs site

### ISSUE THREAD

**Issue by cbachhuber** [rvben/rumdl#863]
### rumdl version

0.2.69

### Installation method

Pre-built binary (via `uvx rumdl@0.2.69`, PyPI wheel)

### Bug description

With `absolute-links = "relative_to_roots"`, a repository-absolute link to an existing directory is accepted without a trailing slash (`/adir`) but rejected with one (`/adir/`).
The equivalent relative link (`../adir/`) is accepted either way.

I understand from [`resolve_under_root_with_opts`](https://github.com/rvben/rumdl/blob/749a7ea75b4bd73c4d8c2392d63c65eddefb490b/src/rules/md057_existing_relative_links.rs#L906) that this is deliberate: a trailing slash marks the destination as a directory-style link, which must serve `index.md`.
Two things still seem worth changing.

**1. The reported message does not describe the actual cause.**
Internally the per-root resolution returns `Resolution::DirectoryWithoutIndex`, but `relative_to_roots` collapses every failure across the roots into:

```
Absolute link '/adir/' was not found under any configured root or the project root
```

The docs-dir mode (`relative_to_docs`) has a precise message for this case — "resolves to directory '…' which has no index.md" — but the roots mode discards it.
The result sends you looking for a missing directory that is in fact present, which is what happened to me.
Even if the rule's verdict stays as it is, surfacing the `DirectoryWithoutIndex` reason when at least one root resolved to a directory would make it actionable.

**2. The `index.md` convention does not hold outside a static-site generator.**
The rationale in the code is MkDocs/Hugo URL routing, where `/section/` serves `section/index.md`.
In a plain repository rendered by GitHub, `/adir/` and `/adir` both resolve to the directory listing, and neither needs an `index.md`.
So under the default (non-MkDocs) flavor, the trailing slash carries no meaning that would justify the stricter check, and it produces a false positive on a link that works.

Related: #632 fixed the no-trailing-slash form of this in v0.1.96 ("MD057 now accepts absolute links that resolve to an existing directory"), which is what makes the remaining trailing-slash difference feel like an inconsistency rather than a deliberate distinction.

### Steps to reproduce

```
/tmp/repro/
├── .rumdl.toml
├── adir/some.md
└── docs/t.md
```

`.rumdl.toml`:

```toml
[MD057]
absolute-links = "relative_to_roots"
roots = ["."]
```

`docs/t.md`:

```markdown
# T

[abs dir no slash](/adir)

[abs dir slash](/adir/)

[rel dir slash](../adir/)
```

Run `rumdl check .` from `/tmp/repro`.

### Expected behavior

Under the default flavor, all three links resolve to the existing `adir/` directory and none is reported.

Failing that: if `/adir/` is to stay an error, the message should name the reason, as `relative_to_docs` already does.

### Actual behavior

```
docs/t.md:5:17: [MD057] Absolute link '/adir/' was not found under any configured root or the project root

Issues: Found 1 issues in 1/2 files (10ms)
```

Only the trailing-slash absolute form is flagged; `/adir` and `../adir/` both pass.

### Configuration

See the `.rumdl.toml` above.

### Operating system

Linux

### Additional context

Concretely, what I would expect is for the `index.md` requirement to follow the flavor rather than the trailing slash:

- `flavor = "mkdocs"` / `absolute-links = "relative_to_docs"`: keep the current behavior, since the site really does route `/section/` to `section/index.md`.
- Default flavor with `relative_to_roots`: treat a trailing slash as filesystem-equivalent, matching how relative links already resolve via `path.exists()`.

If you would rather keep the current semantics, an opt-out (something like `directory-links = "filesystem" | "index"`) would work just as well for our case.
We hit this while replacing lychee with `MD057` in a repository whose docs link to directories such as `/ansible/`; the workaround was to rewrite those links without the trailing slash.

Happy to open a PR for whichever shape you prefer.

**Comment by cbachhuber** [comment]
Fixed and released, thank you for the quick turnaround! 

You might want to look into why `Closes #863` from your commit message didn't close this. Is it capitalization? All documents I found reference lower-case verbs only, but per gemini, the close-via-commit-message logic is case-insensitive.
